Pokemon_master.py

```diff
@@ -0,0 +1,150 @@
+            if self.type == "grass":
+                attack_multiplier = 2
+            if self.type == "water":
+                attack_multiplier = 1
+            if self.type == "fire":
+                attack_multiplier = 0.5
+        if self.opposite_pokemon.type == "fire":
+            if self.type == "grass":
+                attack_multiplier = 0.5
+            if self.type == "water":
+                attack_multiplier = 2
+            if self.type == "fire":
+                attack_multiplier = 1
+
+        attack_damage = self.level * attack_multiplier
+        print(self.trainer_name)
+        print("{name} is dealing {damage} damage to the opposite pokemon".format(name=self.name, damage=attack_damage))
+        return attack_damage
+
+
+class Trainer():
+    def __init__(self, pokemons, trainer_name, num_of_potions, active_pokemon):
+        self.pokemons = pokemons
+        self.trainer_name = trainer_name
+        self.num_of_potions = num_of_potions
+        self.active_pokemon = self.pokemons[active_pokemon -1]
+        self.knocked_out_pokemons = []
+
+        #assigning trainer name to pokemon and classifying which pokemon is knocked out
+        for pokemon in self.pokemons:
+            pokemon.trainer_name = self.trainer_name
+            if pokemon.is_knocked_out == True:
+                self.knocked_out_pokemons.append(pokemon)
+                self.pokemons.remove(pokemon)
+
+
+
+    def use_potion(self):
+        # reduces num_of_potions by 1
+        # heals active pokemon
+        print(self.trainer_name)
+        amount_of_healing = 20
+        self.active_pokemon.regain_health(amount_of_healing)
+        self.num_of_potions += -1
+        print("{trainer_name}'s {pokemon} has been heald by {healing} \n".format(trainer_name=self.trainer_name, pokemon=self.active_pokemon.name, healing=amount_of_healing))
+
+    def attack_trainer(self, trainer_to_attack):
+        # give damage to another trainer's active pokemon
+        if self.active_pokemon.is_knocked_out == True:
+            print("{pokemon} is knocked out and can't fight anymore, choose a different pokemon: {pokemons}\n".format(pokemon=self.active_pokemon.name, pokemons=self.pokemons))
+
+        else:
+            pokemon_we_attack = trainer_to_attack.active_pokemon
+            attack_damage = self.active_pokemon.attack(pokemon_we_attack)
+            pokemon_we_attack.lose_health(attack_damage)
+
+            if pokemon_we_attack.current_health <= 0:
+                pokemon_we_attack.knock_out()
+
+
+
+    def switch_pokemon(self, pokemon_number):
+        print(self.trainer_name)
+        print("Your pokemons are {pokemons}".format(pokemons=self.pokemons))
+        if self.pokemons[pokemon_number - 1].is_knocked_out == True:
+            print("{pokemon} is knocked out and you can't choose it".format(pokemon=self.pokemons[pokemon_number - 1].name))
+        else:
+            self.active_pokemon = self.pokemons[pokemon_number - 1]
```

Comment:
Q:  self.pokemons[pokemon_number - 1]  jeżeli damy 0, to będzie wyjątek czy przypisze jakieś gówno do zmiennej?